include/s3select.h

```diff
@@ -2057,6 +2103,20 @@ class base_s3object
           }
 	}
 
+        if(m_is_limit_on && m_processed_rows == m_limit)
```

Comment:
should check the possibility, of "merge" the `limit` condition with `end-of-stream` flow.
instead of additional code-lines/flow in `getMatchRow`

limit and end-of-stream, both belong to the same flow (end of processing)